src/trio/_path.py

```diff
@@ -32,219 +21,97 @@
         OpenBinaryModeWriting,
         OpenTextMode,
     )
-    from typing_extensions import Concatenate, Literal, ParamSpec, TypeAlias
-
-    from trio._file_io import AsyncIOWrapper as _AsyncIOWrapper
+    from typing_extensions import Concatenate, Literal, ParamSpec
 
     P = ParamSpec("P")
 
-T = TypeVar("T")
-StrPath: TypeAlias = Union[str, "os.PathLike[str]"]  # Only subscriptable in 3.9+
+    Self = TypeVar("Self", bound="Path")
```

Comment:
Why are we creating a `Self` type when `from typing_extensions import Self` exists?

Reply:
A good question! The `Self` type is special and can only be used in conventional class method declarations. Since we're using it here in a decorator/wrapper, it's not usable.

Reply:
I kinda think it should be named differently to avoid confusion with the special type, maybe `PathT`?

Reply:
Calling this sort of thing `Self` was fairly common before the _real_ `Self` type was introduced. But yeah, I'll do that, it makes sense now!